examples/nested_list/src/list.rs

```diff
@@ -62,22 +60,19 @@ pub enum Msg {
 pub struct Props {
     pub children: ChildrenRenderer<ListVariant>,
     pub on_hover: Callback<Hovered>,
-    pub weak_link: WeakComponentLink<List>,
 }
 
 pub struct List {
     inactive: bool,
 }
 
-impl Component for List {
+impl BaseComponent for List {
     type Message = Msg;
     type Properties = Props;
+    type Reference = Scope<Self>;
```

Comment:
Whilst I was originally suggesting to not classify `ref` as a normal prop, however, after @hamza1311 brought up the idea of making the `ref` as a normal prop, I actually think this is a better approach and leads to a much simpler design.

Suppose that if there are 2 refs (e.g.: `ref` and `inputRef`) for a single component, how to handle these 2 at the same time?
(It's not always the case to simply forward the `inputRef` to the inner input element, as it can be enhanced by adding additional methods to it.)
In addition, I don't think it actually provides any benefit of actually separating the handling of these 2 refs especially if you have to set it manually.

Reply:
There's a semantic difference: `ref` means that *this component* is responsible for *binding* the referenced value. Normally, you *get* a currently referenced value from a `HtmlRef`. Hence why `crate` accepts a special `BindableRef` type that allows you to bind (and maybe rebind, I'm not sure at this point, but it could make sense) a value. `BindableRef` also could have a method to decay it to a ref the component binds to one of its children, i.e. to forward the ref.

But being the `HtmlRef` that this and only this component is in charge of binding, is imo special enough to deserve separate treatment.

Reply:
If you wrap a third party component library that is linked with `VRef` and you have 2 props (`ref` and `input_ref`) with `ref` linked to the container div.

How do you expose `input_ref` to the inner input element created by `VRef` without creating another component?

Reply:
Added `BindableRef::forward` so the component can capture a `HtmlRef` it can itself bind on children returned from `view`.

Reply:
I suppose `VRef` should accept a `NodeRef`?

Reply:
> I suppose `VRef` should accept a `NodeRef`, too?

I don't see how this solves the example in previous comment.

`VRef` is registered on `el` but `input_ref` is supposed to be registered on `input_el` which is queried from the `el` by `query_selector`.

Reply:
Ah I didn't quite get what you were asking: The best strategy for this case, imo, would be to expose a struct containing all the referenced elements. Such as

```rust
struct InputRefs {
    el: Element,
    input_el: Element,
}
impl ComponentWithRef for MyInput {
    type Reference = InputRefs;

    fn create(ctx: Context<Self>, bindable_ref: BindableRef<Self>) {
        let el = document().create_element("my-input");
        let input_el = el.query_selector("input");
        bindable_ref.set(InputRefs { el: el.clone(), input_el: input_el.clone() });
        // store el in Self
    }
    fn view(...) {
        VNode::VRef(self.el);
    }
```

Reply:
I forgot to mention that usually a custom element will not render its content until its `connectedCallback` is invoked. Which means that the `input` element will not become available until `rendered()` cycle.